realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -404,6 +404,192 @@ public boolean isValid() {
         return this;
     }
 
+    // In
+
+    /**
+     * In comparison. This allows you to test if objects match any value in an array of values.
+     *
+     * @param fieldName the field to compare.
+     * @param values values to compare with and it cannot be null or empty.
+     * @return the query object.
+     * @throws java.lang.IllegalArgumentException if values is empty or do not match class or field type.
+     */
+    public RealmQuery<E> in(String fieldName, String[] values) {
+        return in(fieldName, values, Case.SENSITIVE);
+    }
+
+    /**
+     * In comparison. This allows you to test if objects match any value in an array of values.
+     *
+     * @param fieldName the field to compare.
+     * @param values values to compare with and it cannot be null or empty.
+     * @param casing how casing is handled. {@link Case#INSENSITIVE} works only for the Latin-1 characters.
+     * @return the query object.
+     * @throws java.lang.IllegalArgumentException if values is empty or do not match class or field type.
+     */
+    public RealmQuery<E> in(String fieldName, String[] values, Case casing) {
+        if (values == null || values.length == 0) {
```

Comment:
Should we allow the 0-length `values` array? It would basically default to the query always returning the empty set, but I would imagine that this `values` array in most cases will be generated in a loop based on some other input, so I think that this array in a lot of cases will actually be empty, making it a hassle for the user to work around.  

Thoughts @realm/java?